# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/JitActivation.cpp
# Commit: 270961083337
# Full Hash: 270961083337349dc71f3a2359b7cdc4b1c83a0f
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-20 14:55:47
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 5: Remove remaining uses of Frame::tls. r=lth
#   
#   Here we remove remaining uses of Frame::tls. There are many places where
#   we use it: in debug frames, in profiling frames, in jit activation, etc.
#   All these places require short fixes to use our new scheme for getting
# ==============================================================================

diff -r db39bf2058e1 -r 270961083337 js/src/vm/JitActivation.cpp
--- a/js/src/vm/JitActivation.cpp	Tue Oct 20 10:24:40 2020 +0000
+++ b/js/src/vm/JitActivation.cpp	Tue Oct 20 10:24:47 2020 +0000
@@ -233,7 +233,7 @@
   void* pc = unwindState.pc;
   const wasm::Frame* fp = wasm::Frame::fromUntaggedWasmExitFP(unwindState.fp);
 
-  const wasm::Code& code = fp->instance()->code();
+  const wasm::Code& code = wasm::GetNearestEffectiveTls(fp)->instance->code();
   MOZ_RELEASE_ASSERT(&code == wasm::LookupCode(pc));
 
   // If the frame was unwound, the bytecodeOffset must be recovered from the